websites/A/AniCrush/presence.ts

```diff
@@ -0,0 +1,212 @@
+			}`;
+
+			presenceData.state = `Episode ${
+				Number(
+					document.querySelector(".active .btn-ep")?.textContent
+				).toLocaleString() || "?"
+			}/${
+				Number(
+					document.querySelector(`.item.item-${streamingType} .name`)
+						?.textContent
+				).toLocaleString() || "?"
+			}`;
+			if (video.exists) {
+				if (await presence.getSetting<boolean>("timestamps")) {
+					presenceData.startTimestamp = startTimestamp;
+					presenceData.endTimestamp = endTimestamp;
+				}
+				if (video.paused) {
+					presenceData.smallImageKey = Assets.Pause;
+					presenceData.smallImageText = "Paused";
+					delete presenceData.startTimestamp;
+					delete presenceData.endTimestamp;
+				} else {
+					presenceData.smallImageKey = Assets.Play;
+					presenceData.smallImageText = "Playing";
+				}
+			}
+			break;
+		}
+		case pathname.startsWith("/detail/"): {
+			presenceData.details = `Viewing ${
+				document.querySelector(
+					".about-anime .infor .heading .main .heading-md.mb-0.text-white"
+				)?.textContent || "?"
+			}`;
+			presenceData.state = `Rating ${
+				document.querySelector(".score .s-mark")?.textContent || "?"
+			}/10.0`;
+			break;
+		}
+		case pathname.startsWith("/watch2gether/create/"): {
+			presenceData.details = `Watch2gether ${
+				document.querySelector(".anime-name a")?.textContent || "?"
+			}`;
+			presenceData.state = "Creating room";
+			break;
+		}
+		case pathname.startsWith("/watch2gether/"): {
+			const streamingType = document
+					.querySelector(".live-header .other-items")
+					.children[1].querySelector(".text")
+					?.textContent?.toLowerCase(),
+				[startTimestamp, endTimestamp] = presence.getTimestamps(
+					Math.floor(video.currentTime),
+					Math.floor(video.duration)
+				),
+				totalEpisodes = Number(
+					Array.from(
+						document.querySelectorAll(`.item.item-${streamingType} .text`)
+					).at(-1)?.textContent
+				),
+				viewers = Number(
+					document.querySelector(".item-viewers span")?.textContent[0]
+				);
+			presenceData.details = `Watch2gether ${
+				document.querySelector(".anime-block .main h3 a")?.textContent || "?"
+			}`;
+			presenceData.state = `Episode ${
+				Number(
+					document
+						.querySelector(".other-items .item-ep span")
+						?.textContent?.replace("Episode ", "")
+				).toLocaleString() || "?"
+			}/${
+				isNaN(totalEpisodes) || !totalEpisodes
+					? "?"
+					: totalEpisodes.toLocaleString() || "?"
+			} | ${
+				isNaN(viewers) || !viewers
+					? "? viewers"
+					: viewers > 1
+					? `${viewers.toLocaleString()} viewers`
+					: `${viewers} viewer`
+			}`;
+			const buttonAllowed = await presence.getSetting<boolean>(
+				"watch2getherJoinRoomButton"
+			);
+			if (buttonAllowed && showButtons) {
+				presenceData.buttons = [
+					{
+						label: "Join Room",
+						url: href,
+					},
+				];
+			}
+			const showTimestamps = await presence.getSetting<boolean>("timestamps");
```

Comment:
For performance and conciseness, try fetching all the settings at the top of the function.